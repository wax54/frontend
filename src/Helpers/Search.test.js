import { render, fireEvent } from '@testing-library/react';
import Search from './Search';

describe('Search Tests', ()=>{
    test("doesn't blow up", () => {
        render(<Search search={() => undefined}/>);
    });

    test("matchesSnapshot", () => {
        const { asFragment } = render(<Search search={ () => undefined }/>);
        expect(asFragment()).toMatchSnapshot();
    });

    test("matchesSnapshot", () => {
        let result = false;
        const { getByPlaceholderText } = render(<Search search={(term) => result=term} />);
        const searchBox = getByPlaceholderText('Search');

        expect(result).toBe(false);

        fireEvent.change(searchBox, { target: { value: 'test' } });

        expect(result).toEqual('test');
    });
});
