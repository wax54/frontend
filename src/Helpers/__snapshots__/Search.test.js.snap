// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Tests matchesSnapshot 1`] = `
<DocumentFragment>
  <form
    class="Search"
  >
    <input
      name="search"
      placeholder="Search"
      type="text"
      value=""
    />
  </form>
</DocumentFragment>
`;
